### PR TITLE
Enable crosscompilation for dump and copy plugins

### PR DESCRIPTION
**Pull request by ehbello**
This pull request enables crosscompilation for dump and copy plugins to be able to use them, for example, providing blobs after build a kernel snap for a different architecture.

### CODE REVIEW

File: snapcraft/tests/test_plugin_copy.py [hunk 1]
```diff
@@ -332,6 +332,10 @@ def test_copy_symlinks_that_should_be_followed(self):
                 with open(destination, 'r') as f:
                     self.assertEqual(f.read(), symlink['expected_contents'])
 
+    def test_copy_enable_cross_compilation(self):
+        c = CopyPlugin('copy', self.mock_options, self.project_options)
+        c.enable_cross_compilation()
```
Comment (sergiusens) [inline snapcraft/tests/test_plugin_copy.py]: this test does not verify anything. Maybe call `.build()` after or add a code comment about the expected behavior

File: snapcraft/tests/test_plugin_dump.py [hunk 2]
```diff
@@ -194,3 +194,7 @@ def test_dump_broken_symlink(self):
             str(raised.exception),
             '{!r} is a broken symlink pointing outside the snap'.format(
                 os.path.join(plugin.builddir, 'bad_relative')))
+
+    def test_dump_enable_cross_compilation(self):
+        plugin = DumpPlugin('dump', self.options, self.project_options)
+        plugin.enable_cross_compilation()
```
Comment (sergiusens) [inline snapcraft/tests/test_plugin_dump.py]: ditto

File: snapcraft/tests/test_plugin_copy.py (hunk 1 above)
Comment (ehbello) [inline snapcraft/tests/test_plugin_copy.py]: This test verifies that this plugin does not stop when the target architecture is different from host architecture. Both plugins do the same on all architectures. The other features already have their own tests.

Comment (ehbello) [inline snapcraft/tests/test_plugin_copy.py]: ... so this test, simply checks that 'enable_cross_compilation' method don't raise any exception.

**Comment by sergiusens** [comment]
ok to test

**Comment by sergiusens** [comment]
LP: #1624544

**Pull request merged**